lib/utils.js

```diff
@@ -438,14 +440,14 @@ const executeOperation = (topology, operation, args, options) => {
 
   // Return a Promise
   if (args[args.length - 1] != null) {
-    console.dir(operation.name);
     throw new TypeError('final argument to `executeOperation` must be a callback');
   }
 
   return new Promise(function(resolve, reject) {
     args[args.length - 1] = (err, r) => {
-      if (session) {
+      if (session && !options.returnsCursor) {
         session.endSession(() => {
+          opOptions.session = undefined;
```

Comment:
Is there a reason that we are not using `delete` here?

Reply:
force of habit from old node optimizing days I guess. `delete` will change the object shape, and require V8 to rebuild the shadow class for the object - but I'm pretty sure this has been fixed with TurboFan.  I'll go ahead and change to `delete` since I agree it's more explicit, and correct.